### PR TITLE
Moving to Nuke.build as build system for dotnet/.net

### PR DESCRIPTION
**Pull request by driekus77**
Notes:
- For parsing release notes I "borrowed" source code from Cake sources. The nuke ChangeLog parsing gave exceptions.
- Could not test (Pre)Publish because of failing UnitTests.

# Types of Changes

## Prerequisites

Please make sure you can check the following two boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project

## Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [x] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

## Description

[Place a meaningful description here.]

### CODE REVIEW

**Review by FlorianRappl — COMMENTED** [review]
OK in general looks great. Should we also remove the CAKE files etc. ?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -69,7 +69,13 @@ jobs:
     steps:
     - uses: actions/checkout@v2
 
-    - uses: actions/setup-dotnet@v1
+    - name: Setup .NET 6.0
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 6.0.x
+
+    - name: Setup .NET 7.0
```
Comment (FlorianRappl) [inline .github/workflows/ci.yml]: No need to run this again. Multiple versions can be supplied (just pipe and use a new line for each version), see https://github.com/actions/setup-dotnet#usage.

File: nuke/Build.cs [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+                Version = $"{Version}-alpha-{buildNumber}";
+            }
+        }
+
+        Log.Information("Building version: {Version}", Version);
+
+        TargetProject = Solution.GetProject(SourceDirectory / TargetProjectName / $"{TargetProjectName}.Core.csproj" );
+        TargetProject.NotNull("TargetProject could not be loaded!");
+
+        TargetFrameworks = TargetProject.GetTargetFrameworks();
```
Comment (FlorianRappl) [inline nuke/Build.cs:103]: Beforehand we've been quite explicit about this one. Will this also discard frameworks on Linux and have all the ones we desire being built?

**Comment by driekus77** [comment]
> OK in general looks great. Should we also remove the CAKE files etc. ?

Removed the 2 files.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (driekus77) [reply .github/workflows/ci.yml]: Fixed in newest to push code.

File: nuke/Build.cs (hunk 2 above)
Comment (driekus77) [reply nuke/Build.cs:103]: Build for Windows:
image

Build for Linux:
image

**Review by FlorianRappl — APPROVED** [review]
LGTM - let's ship it!

**Pull request merged**